### PR TITLE
Bump Mermaid JS to v10.6.1

### PR DESCRIPTION
**Pull request by Gauravpadam**
An attempt/step forward for #31960

This doesn't fix the issue completely, But I can see some diagrams rendering from it
This also removes the second instance of the script from `layouts/partials/scripts.html`

### CODE REVIEW

File: layouts/partials/scripts.html [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <script src="/js/popper-1.16.1.min.js" intregrity="sha384-9/reFTGAW83EW2RDu2S0VKaIzap3H66lZH81PoYlFhbGU+6BZp6G7niu735Sk7lN" crossorigin="anonymous"></script>
 <script src="/js/bootstrap-4.6.1.min.js" integrity="sha384-VHvPCCyXqtD5DqJeNxl2dtTyhF78xXNXdkwX1CZeRusQfRKp+tA7hAShOK/B/fQ2" crossorigin="anonymous"></script>
 <script src="{{ "js/script.js" | relURL }}"></script>
-<script async src="/js/mermaid-8.13.4.min.js" integrity="sha384-5hHNvPeMrNH14oM3IcQofDoBhiclNK3g2+hnEinKzQ07C4AliMeVpnvxuiwEGpaO" crossorigin="anonymous"></script>
+<script async src="/js/mermaid-10.6.1.min.js" integrity="sha384-5hHNvPeMrNH14oM3IcQofDoBhiclNK3g2+hnEinKzQ07C4AliMeVpnvxuiwEGpaO" crossorigin="anonymous"></script>
```
Comment (sftim) [inline layouts/partials/scripts.html:5]: I think the `integrity` value should change too.

Comment (Gauravpadam) [reply layouts/partials/scripts.html]: Generated a new hash using OpenSSL

File: static/js/mermaid-10.6.1.min.js
Comment (sftim) [inline static/js/mermaid-10.6.1.min.js:1]: - How can we (reviewers) verify that this code is authentic?
- How should we verify the updated hash?

I think I can work those out, but a hint is helpful for sure.

Comment (Gauravpadam) [reply static/js/mermaid-10.6.1.min.js:1]: Sure, we can authenticate the script using the hash itself. I'll write down some steps:

- We can send a curl request to `https://unpkg.com/mermaid@10.6.1/dist/mermaid.min.js` which is the source of the script
- I've used the hash function `shasum -b -a 384 FILENAME.js | awk '{ print $1 }' | xxd -r -p | base64` to generate the hash

If the hash on my code and the generated hash matches, the script is authentic and so is the hash

Alternatively, `https://www.srihash.org` does the same and is more efficient

File: layouts/partials/head.html [hunk 2]
```diff
@@ -75,8 +75,8 @@
 <meta property="og:type" content="{{ $ogType }}">
 
 {{ if .HasShortcode "mermaid" }}
-<!-- Copied from https://unpkg.com/mermaid@8.13.4/dist/mermaid.min.js -->
-<script async src="{{ "js/mermaid-8.13.4.min.js" | relURL }}"></script>
+<!-- Copied from https://unpkg.com/mermaid@10.6.1/dist/mermaid.min.js -->
+<script async src="{{ "js/mermaid-10.6.1.min.js" | relURL }}"></script>
```
Comment (sftim) [inline layouts/partials/head.html:79]:
```suggestion
<script async src="/js/mermaid-10.6.1.min.js" integrity="sha384-+NGfjU8KzpDLXRHduEqW+ZiJr2rIg+cidUVk7B51R5xK7cHwMKQfrdFwGdrq1Bcz" crossorigin="anonymous"></script>
```

File: layouts/partials/scripts.html [hunk 3]
```diff
@@ -2,7 +2,7 @@
 <script src="/js/popper-1.16.1.min.js" intregrity="sha384-9/reFTGAW83EW2RDu2S0VKaIzap3H66lZH81PoYlFhbGU+6BZp6G7niu735Sk7lN" crossorigin="anonymous"></script>
 <script src="/js/bootstrap-4.6.1.min.js" integrity="sha384-VHvPCCyXqtD5DqJeNxl2dtTyhF78xXNXdkwX1CZeRusQfRKp+tA7hAShOK/B/fQ2" crossorigin="anonymous"></script>
 <script src="{{ "js/script.js" | relURL }}"></script>
-<script async src="/js/mermaid-8.13.4.min.js" integrity="sha384-5hHNvPeMrNH14oM3IcQofDoBhiclNK3g2+hnEinKzQ07C4AliMeVpnvxuiwEGpaO" crossorigin="anonymous"></script>
+<script async src="/js/mermaid-10.6.1.min.js" integrity="sha384-+NGfjU8KzpDLXRHduEqW+ZiJr2rIg+cidUVk7B51R5xK7cHwMKQfrdFwGdrq1Bcz" crossorigin="anonymous"></script>
```
Comment (sftim) [inline layouts/partials/scripts.html:5]:
```suggestion
```
Shouldn't we _drop_ this?

**Review by sftim — COMMENTED** [review]
~/lgtm~

Wait wait wait.

Why do we have the Mermaid script included twice?

Well, let's upgrade it, but also let's have another PR that makes sure we only ever load it **once** (ideally, only when there's a Mermaid shortcode on the page).

/lgtm

**Comment by Gauravpadam** [comment]
Let's make it a clean bump in this PR itself, I'd like to focus on the actual fix in the follow-up PRs

File: layouts/partials/scripts.html (hunk 3 above)
Comment (Gauravpadam) [reply layouts/partials/scripts.html]: I'd drop this sooner than in later PRs

**Comment by sftim** [comment]
If dropping MermaidJS from the second location, please tweak the PR description to make that clear.

**Comment by sftim** [comment]
https://deploy-preview-44644--kubernetes-io-main-staging.netlify.app/docs/contribute/ LGTM

/lgtm

**Comment by reylejano** [comment]
Thank you for addressing comments
Preview of pages with diagrams are working
/approve

**Pull request merged**